pkg/api/controller.go

```diff
@@ -802,27 +810,71 @@ func (c *Controller) ListUsers(w http.ResponseWriter, r *http.Request, params Li
 	writeResponse(w, http.StatusOK, response)
 }
 
+func (c *Controller) generateResetPasswordToken(email string) (string, error) {
+	secret := c.Auth.SecretStore().SharedSecret()
+	currentTime := time.Now()
+	return GenerateJWTResetPassword(secret, email, currentTime, currentTime.Add(DefaultResetPasswordExpiration))
+}
+
+func (c *Controller) inviteUserRequest(email string) error {
+	// TODO (@shimi9276) create template for sending the email with link for invite using the passed template
+	token, err := c.generateResetPasswordToken(email)
+	if err != nil {
+		return err
+	}
+	err = c.Emailer.SendEmailWithLimit([]string{email}, token, token, nil)
+	if err != nil {
+		return err
+	}
+	c.Logger.WithField("email", email).Info("invite email sent")
+	return nil
+}
+
 func (c *Controller) CreateUser(w http.ResponseWriter, r *http.Request, body CreateUserJSONRequestBody) {
+	invite := swag.BoolValue(body.InviteUser)
+	id := body.Id
+	var email *string
+	if invite {
+		addr, err := mail.ParseAddress(body.Id)
+		if err != nil {
+			c.Logger.WithError(err).WithField("userID", id).Warn("failed parsing email")
+			writeError(w, http.StatusBadRequest, "Invalid email format")
+			return
+		}
+		id = strings.ToLower(addr.Address)
+		email = &body.Id
```

Comment:
Not sure we need to do that, because if it is an invalid email address it would have failed in the previous step (by the parsing). In any case, passed &addr.Address which is definitely a valid email.

Reply:
I believe @nopcoder refers to the difference between "addr-spec" (what you are looking for) and "address" or "mailbox" (what the code parses).  See links on my looooong comment appearing soon below.